### ISSUE TITLE
Using TochMetrics for code simplification

### ISSUE THREAD

**Issue by Borda** [pyg-team/pytorch_geometric#4283]
### 🛠 Proposed Refactor

We have been developing [TochMetrics](https://github.com/PyTorchLightning/metrics) to be a general-purpose metric as well some domain-specific use-cases. In many cases, we have an exact mapping to `scikit-learn` with verification/testing to the reference metric for its correctness. The TM includes functional as well as `nn.Module` versions and for the most/standard metrics, the only dependency is Pytorch (for the domain-specific metrics you need to install related extras). 

### Suggest a potential alternative/fix

WIth using [TM metrics](https://torchmetrics.readthedocs.io/en/stable/) you may rely on the widely tested correctness (testing against good standard `scikit-learn` in multiple OS environments and all PyTorch versions above v1.4) and later you can use `nn.Module` to leverage update and compute.

Overall if you are fine with it we are happy to draft a PR with a suggested change to verify in place the impact.
If you have any questions, happy to follow up with me or @tchaton

What I have quickly checked, al this [`torch_geometric.utils.metric`]( https://github.com/pyg-team/pytorch_geometric/blob/master/torch_geometric/utils/metric.py) can be replaced by TM

**Comment by rusty1s** [comment]
Thanks for this great issue. We are already using `torchmetrics` in all our PL examples, and so I am happy to make the switch to it completely. `torch_geometric.utils.metric` is somewhat old and not well maintained. In fact, as far as I can see, only the `intersection_and_union` metric is actually used inside PyG and `examples/`.

As such, we could:
* remove `torch_geometric.utils.metric` completely
* Update the examples to make use of `torchmetrics`

Let me know what you think.

**Comment by Borda** [comment]
cool, thank you @rusty1s! I have started a draft PR as I may have a few additional questions to deliver the best...
so let's move the additional/implementation discussion to the open #4287
